squirrel.frontier/src/main/java/org/dice_research/squirrel/frontier/impl/FrontierImpl.java

```diff
@@ -60,6 +63,14 @@
      * SPARQL, DEREFERENCEABLE or UNKNOWN
      */
     protected UriProcessor uriProcessor;
+    /**
+     * {@link UriGenerator} used to generate additional domain variants of a URI
+     */
+    protected UriGenerator domainUriGenerator;
+    /**
+     * {@link UriGenerator} used to generate additional well-known VoID variant of a URI
+     */
+    protected UriGenerator wellKnownUriGenrator;
```

Comment:
This could be made more generic. Please have a look at the functionality the Spring framework offers: https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/beans/factory/annotation/Autowired.html  (especially the part about collections).